lib/src/ui/pages/profile/profile_notifier.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:allthenews/src/domain/authorization/authentication_repository.dart';
+import 'package:allthenews/src/domain/communication/firebase_exception.dart';
+import 'package:allthenews/src/ui/pages/authentication/authentication_message_provider.dart';
+import 'package:allthenews/src/ui/pages/profile/profile_state.dart';
+import 'package:flutter/widgets.dart';
+
+class ProfileNotifier extends ChangeNotifier {
+  final AuthenticationRepository _authorizationRepository;
+  final AuthenticationMessageProvider _authenticationMessageProvider;
+  ProfileState _state = const ProfileState();
+
+  ProfileNotifier(
+    this._authorizationRepository,
+    this._authenticationMessageProvider,
+  );
+
+  ProfileState get state => _state;
+
+  VoidCallback returnToProfile;
+
+  void initUserState() {
+    _setNotifierState(_state.copyWithLoading(isLoading: true));
+    _authorizationRepository.observeUserChanges().listen((user) {
+      _setNotifierState(_state.copyWithUserAndLoading(user: user, isLoading: false));
```

Comment:
w sumie może tu wystąpić błąd, i zastanawiam się czy stream może nic nie wysłać, wtedy też będzie wieczny loading

Reply:
jeżeli nie będzie usera to zwróci nulla, każda inna zmiana będzie userem. Jeżeli chodzi o błędy to dokumentacja nic nie mówi na ten temat

Reply:
a choćby brak neta?

Reply:
hmm sprawdzę

Reply:
jak nie ma neta to user leci jako null

Reply:
ciekawe :D w takim razie jest git :D

Reply:
w sumie słabo, bo jak rozpoznać czy jestem niezalogowany czy po prostu nie mam neta? dopiero przy logowaniu się dowiem

Reply:
hmm... racja, zostawiłbym to na osobne zadanie, bo chyba musi być na tym ekranie jakiś dodatkowy check robiony 🤔

Reply:
dodałem do trello